instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/HttpClientTracer.java

```diff
@@ -182,9 +182,13 @@ private void setUrl(AttributeSetter setter, REQUEST request) {
     try {
       URI url = url(request);
       if (url != null) {
+        if (url.getUserInfo() != null) {
+          throw new IllegalArgumentException("http.url MUST NOT contain credentials!");
```

Comment:
Hi @Hangzhi - thanks for the contribution! We don't want to cause an application error due to the presence of credentials, only that we don't record it. Below we call `url.toString` and we just need to make sure that is called with a url that has had the user info removed.

Also, we will need a unit test for the behavior. Here we have many different urls /expected url

https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation-api/src/test/groovy/io/opentelemetry/instrumentation/api/tracer/HttpClientTracerTest.groovy#L73

So you can add a test which adds a url with credentials and an expected url without.

Reply:
Thanks for your review! I'll work on that.

Reply:
@aantono Hello aantono. I removed the user info and added a test according to your comment. I got this [failing test
](https://github.com/open-telemetry/opentelemetry-java-instrumentation/runs/2276486894?check_suite_focus=true#step:7:3066), but I did some search but still cannot figure out how to fix this. Could you please give me some more hints on how to solve this problem?

Reply:
hey @Hangzhi, this is a known sporadic error that will be fixed by #2728

> should suppress nested CLIENT span if already under parent CLIENT span

Reply:
I hit re-run on the tests for now